File: frontend/src/app/modules/api/models/forms/vote-form.ts
export interface VoteForm {
  id: string;
  title: string;
  description: string;
  createdAt: Date;
  createdBy: string;
  isCompleted: boolean;
  questions: {
    id: string;
    text: string;
    isMultipleAllowed: boolean;
    isCustomAllowed: boolean;
    options: {
      id: string;
      value: string;
    }[];
  }[];
}

File: backend/VotingApp.Web/Models/Entities/VoteForm.cs
using System.Collections;

namespace VotingApp.Web.Models.Entities;

public class VoteForm
{
    public VoteFormId Id { get; set; } = Guid.NewGuid();
    
    public string Title { get; set; } = default!;
    
    public string Description { get; set; } = default!;
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public UserId CreatedBy { get; set; } = default!;

    public bool IsCompleted { get; set; } = false;

    public VoteResultsAccess Access { get; set; } = default!;

    public ICollection<VoteQuestion> Questions { get; set; }
        = new List<VoteQuestion>();

    public ICollection<VoteResult> Results { get; set; } 
        = new List<VoteResult>();

    public ICollection<VoteFormComment> Comments { get; set; }
        = new List<VoteFormComment>();
}